### ISSUE TITLE
(Accessibility) background are still tabbable when modal is open

### ISSUE THREAD

**Issue by changLiuUNSW** [ng-bootstrap/ng-bootstrap#1363]
When modal dialog is open, the elements in the background are still tabbable. Also ```aria-hidden=true``` should add to all other elements except modal itself.

I have checked ```ui-bootstrap``` (angular 1). The modal does not have above issues

**Comment by pkozlowski-opensource** [comment]
We are aware of this issue and we've got already a ticket tracking this: #642. The good news is that accessibility is a no 1 priority for few next releases so we should be able to tackle it sooner than later.

Closing as a duplicate of #642

**Comment by changLiuUNSW** [comment]
Thanks for your quick response. Hope you guys can fix it soon, because we (bank) take accessibility very seriously when we choose the third party library. 
Looks like material 2 already solved this issue with ```focus-trap```:
https://github.com/angular/material2/blob/master/src/lib/core/a11y/focus-trap.ts
I may create a PR for you guys if you do not mind.

**Comment by pkozlowski-opensource** [comment]
> I may create a PR for you guys if you do not mind

PRs are more than appreciated. And just to be clear: this library is taking accessibility _very_ seriously: we want to have World-class accessibility support, we've got experienced people and we work towards this goal. This doesn't change the fact that having World-class accessibility support takes time and dedicated effort. So once again, all PRs are more than welcomed, provided that those come with unit tests. Thnx!

**Comment by changLiuUNSW** [comment]
@pkozlowski-opensource  
I am trying to create a PR to add ```focus-trap```, but I am not sure whether it is OK to copy it from ```ng-material 2```.

Also do I need to create a new module like ```A11yModule``` and put it under folder ```a11y``` (to be created)? Or create a ````core module```` and put ```A11yModule``` in it (same structure like ```ng-material 2```)

**Comment by changLiuUNSW** [comment]
@pkozlowski-opensource  Just created PR. Please have a review.
https://github.com/ng-bootstrap/ng-bootstrap/pull/1464
